Connectors/Revit/Speckle.Connectors.RevitShared/Operations/Send/RevitRootObjectBuilder.cs

```diff
@@ -52,92 +50,92 @@ ILogger<RevitRootObjectBuilder> logger
+    foreach (var id in objects)
+    {
+      var el = _conversionContextStack.Current.Document.GetElement(id);
+      if (el != null)
       {
-        var el = _conversionContextStack.Current.Document.GetElement(id);
-        if (el != null)
-        {
-          revitElements.Add(el);
-        }
+        revitElements.Add(el);
       }
+    }
 
-      if (revitElements.Count == 0)
-      {
-        throw new SpeckleSendFilterException("No objects were found. Please update your send filter!");
-      }
+    if (revitElements.Count == 0)
+    {
+      throw new SpeckleSendFilterException("No objects were found. Please update your send filter!");
+    }
+
+    List<SendConversionResult> results = new(revitElements.Count);
 
-      // Unpack groups (& other complex data structures)
-      var atomicObjects = _elementUnpacker.UnpackSelectionForConversion(revitElements).ToList();
+    // Unpack groups (& other complex data structures)
+    var atomicObjects = _elementUnpacker.UnpackSelectionForConversion(revitElements).ToList();
 
-      var countProgress = 0;
-      var cacheHitCount = 0;
-      List<SendConversionResult> results = new(revitElements.Count);
+    var countProgress = 0;
+    var cacheHitCount = 0;
 
-      foreach (Element revitElement in atomicObjects)
+    foreach (Element revitElement in atomicObjects)
+    {
+      ct.ThrowIfCancellationRequested();
+      string applicationId = revitElement.UniqueId;
+      string sourceType = revitElement.GetType().Name;
+      try
       {
-        ct.ThrowIfCancellationRequested();
-        string applicationId = revitElement.UniqueId; // NOTE: converter set applicationIds to unique ids; if we ever change this in the converter, behaviour here needs to match.
-        string sourceType = revitElement.GetType().Name;
-        try
+        Base converted;
+        if (_sendConversionCache.TryGetValue(sendInfo.ProjectId, applicationId, out ObjectReference? value))
         {
-          Base converted;
-          if (_sendConversionCache.TryGetValue(sendInfo.ProjectId, applicationId, out ObjectReference? value))
-          {
-            converted = value;
-            cacheHitCount++;
-          }
-          else
-          {
-            converted = _converter.Convert(revitElement);
-            converted.applicationId = applicationId;
-          }
-
-          var collection = _sendCollectionManager.GetAndCreateObjectHostCollection(revitElement, _rootObject);
-          collection.elements.Add(converted);
-          results.Add(new(Status.SUCCESS, applicationId, sourceType, converted));
+          converted = value;
+          cacheHitCount++;
         }
-        catch (Exception ex) when (!ex.IsFatal())
+        else
         {
-          _logger.LogSendConversionError(ex, sourceType);
-          results.Add(new(Status.ERROR, applicationId, sourceType, null, ex));
+          converted = RevitTask.RunAsync(() => _converter.Convert(revitElement)).GetAwaiter().GetResult();
```

Comment:
This is also blocking a thread.  Why do this?

Reply:
Jedd get them back